js/tests.js

```diff
@@ -1,5 +1,186 @@
-QUnit.test("hello test", function(assert) {
-  assert.strictEqual(1 + 1, 2, "One plus one is two");
+var attack;
+var counterAttack;
+var x;
+var y;
+
+QUnit.test('hello test', function(assert) {
+  assert.strictEqual(1 + 1, 2, 'One plus one is two');
+});
+//__________________________
+
+// TESTS FOR SOLUTION PART 2
+//__________________________
+
+// Testing Constructor
+
+QUnit.module ('superHero.constructor');
+test('constructor', function(assert) {
+
+	var m = new SuperHero('testHero');
+
+	var hasName = m.superName;
+	assert.strictEqual(hasName, 'testHero', 'has property superName');
+
+	var hasPower = m.hasSecretPower;
+	assert.strictEqual(hasPower, true, 'has property secretPower');
+
+	var isAlive = m.isAlive;
+	assert.strictEqual(isAlive, true, 'has property isAlive');
+
+	var hasWeapon = m.hasWeapon;
+	assert.strictEqual(hasWeapon, true, 'has property hasWeapon');
+
+	var hasShield = m.hasShield;
+	assert.strictEqual(hasShield, true, 'has property hasShield');
+
+});
+
+// Testing Child Class Constructor
+
+QUnit.module ('superChild.constructor');
+test('constructor', function(assert) {
+
+	var k = new SuperChild('testChild');
+
+	var hasName = k.superName;
+	assert.strictEqual(hasName, 'testChild', 'has property superName');
+
+	var hasBomb = k.hasBomb;
+	assert.strictEqual(hasBomb, true, 'has bomb');
+
+});
+
+var a = 'My battle skills combined with these weapons will do you in!!';
+var b = 'I will fight you with my bare hands!!';
+var c = 'With these secret powers you don\'t have a chance!!';
+var d = 'With this shield any weapon you try is just a toy!!';
+var e = 'Ha! Your attack is no match for my secret powers!!';
+var f = 'GOOD AND EVIL BATTLE TO THE END!!!';
+var g = 'EVIL HAS TRIUMPHED -- BUT IT WILL NEVER ENDURE!!';
+var h = 'GOOD TRUMPS EVIL AGAIN!!';
+var j = "NOT SO FAST! THIS BOMB IS THE END TO ALL YOUR EVIL";
+
+//Testing attack method with one instance of superHero: 'testHero'
+
+QUnit.module ('superHero.prototype.attack');
+QUnit.test('method attack', function(assert) {
+
+	var m = new SuperHero('testHero');
+	attack = m.attack();
+
+
+	if
+		(m.hasBomb === true) {
+			assert.strictEqual(attack, j, 'NOT SO FAST! THIS BOMB IS THE END TO ALL YOUR EVIL');
+
+	} else if
+
+		(m.hasSecretPower === false && m.hasWeapon === true) {
+			assert.strictEqual(attack, a, 'My battle skills combined with these weapons will do you in!!');
+
+	} else if
+
+		(m.hasSecretPower === false && m.hasWeapon === false) {
+			assert.strictEqual(attack, b, 'I will fight you with my bare hands!!');
+
+	} else if
+
+		(m.hasSecretPower === true && m.hasWeapon === true) {
+			assert.strictEqual(attack, c, 'With these secret powers you don\'t have a chance!!');
+
+	} else
+
+		{assert.strictEqual(attack, c, 'With these secret powers you don\'t have a chance!!');
+	}
```

Comment:
Hmm, yeah, tough to test random behavior. When that's the case, you will often see people [stubbing](https://en.wikipedia.org/wiki/Mock_object) methods that have non-deterministic behavior (using a library like [Sinon](http://sinonjs.org/)), so that their tests _can_ be deterministic.